tests/Doctrine/Tests/Persistence/Mapping/FileDriverTest.php

```diff
@@ -1,164 +1,202 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Doctrine\Tests\Persistence\Mapping;
 
 use Doctrine\Persistence\Mapping\ClassMetadata;
 use Doctrine\Persistence\Mapping\Driver\FileDriver;
 use Doctrine\Persistence\Mapping\Driver\FileLocator;
 use Doctrine\Tests\DoctrineTestCase;
+use PHPUnit\Framework\MockObject\MockObject;
 use function strpos;
 
 class FileDriverTest extends DoctrineTestCase
 {
-    public function testGlobalBasename()
+    public function testGlobalBasename() : void
     {
-        $driver = new TestFileDriver([]);
+        $driver = $this->createTestFileDriver([]);
 
         self::assertNull($driver->getGlobalBasename());
 
         $driver->setGlobalBasename('global');
         self::assertSame('global', $driver->getGlobalBasename());
     }
 
-    public function testGetElementFromGlobalFile()
+    public function testGetElementFromGlobalFile() : void
     {
-        $driver = new TestFileDriver($this->newLocator());
+        $driver = $this->createTestFileDriver($this->newLocator());
+
         $driver->setGlobalBasename('global');
 
         $element = $driver->getElement('stdGlobal');
 
-        self::assertSame('stdGlobal', $element);
+        self::assertSame($driver->stdGlobal, $element);
```

Comment:
Can you explain this change pls? This change made this test case a bit confusing to me, as it's more dynamic and assertion depends on driver behaviour.

Reply:
The test before was setting up the mock to return strings on a method that should return a ClassMetadata. PHP just didn't care because of the lack of type hints. As soon as we had proper type hints, the test failed, so this is how I fixed it.